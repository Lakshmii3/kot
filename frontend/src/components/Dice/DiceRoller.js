import React from 'react';
import "./DiceRoller.css";

import DiceBoard from './DiceBoard'

var rollExample = [
  'heart',
  'energy',
  'paw',
  '1',
  '2',
  '3'
];

class DiceRoller extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      rolledDice: rollExample, // this holds the name of each list
    };

    this.RequestRoll = this.RequestRoll.bind(this);
  }

  async FetchRoll() {
    try {
      const res = await fetch('http://127.0.0.1:8000/api/dice/');

      const json_data = await res.json();
      var result = [];

      result.push([0, json_data.dice1]);
      result.push([1, json_data.dice2]);
      result.push([2, json_data.dice3]);
      result.push([3, json_data.dice4]);
      result.push([4, json_data.dice5]);
      result.push([5, json_data.dice6]);
      result.push([6, json_data.dice7]);

      // for (var i in json_data)
      //   result.push([i, json_data[i]]);
      return result;
    } catch (err) {
      console.log(err);
      //alert(err); // TypeError: failed to fetch
    }
  }

  async RequestRoll(e) {

    // let data = rollExample;

    try {

      let result = await this.FetchRoll();

      this.setState({ rolledDice: result });
    } catch (e) {
      console.log(e);
    }

    // TO DO : Call dice REST endpoint requesting reroll.
    //this.props.sendMessage(this.state.message)

    // this.setState({
    //   rolledDice: data
    // })
  }

  render() {
    return (
      <div className="DiceRoller">
        <DiceBoard data={this.state.rolledDice} />
        <button onClick={this.RequestRoll}>Roll</button>
      </div>
    )
  }
}

export default DiceRoller;