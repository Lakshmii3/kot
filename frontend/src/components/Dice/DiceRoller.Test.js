import React from 'react';
import { shallow } from 'enzyme';

import DiceRoller from './DiceRoller';

describe('Verify DiceRoller layout template', () => {
    it('DiceRoller should render correctly', () => {
        const errorPage = shallow(<DiceRoller />);
        expect(errorPage).toMatchSnapshot();
    });
})