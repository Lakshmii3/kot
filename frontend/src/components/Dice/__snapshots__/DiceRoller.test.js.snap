// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify DiceRoller layout template DiceRoller should render correctly 1`] = `
<div
  className="DiceRoller"
>
  <DiceBoard
    callbackSendDiceSelectionOut={[Function]}
    data={Array []}
  />
  <button
    disabled={false}
    onClick={[Function]}
  >
    Roll
  </button>
</div>
`;
